### PR TITLE
Support Federation Spec 2.7

### PR DESCRIPTION
**Pull request by TygerTaco**
Update federation support for v2.7 which includes the `@authenticated`, `@requiresScopes`, `@policy` directives, as well as the `label` argument for `@override`. 

## Description

Update federation support to be in alignment with the apollo specification [here](https://www.apollographql.com/docs/federation/subgraph-spec#subgraph-schema-additions)

Specifically:
```
directive @authenticated on FIELD_DEFINITION | OBJECT | INTERFACE | SCALAR | ENUM
directive @requiresScopes(scopes: [[federation__Scope!]!]!) on FIELD_DEFINITION | OBJECT | INTERFACE | SCALAR | ENUM
directive @policy(policies: [[federation__Policy!]!]!) on FIELD_DEFINITION | OBJECT | INTERFACE | SCALAR | ENUM

directive @override(from: String! lable: string) on FIELD_DEFINITION
```

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

Update federation support for v2.6 which includes the @authenticated, @requiresScopes, and @policy directives.

```
    @strawberry.federation.type(
        authenticated=True,
        policy=[["client", "poweruser"], ["admin"]],
        requires_scopes=[["client", "poweruser"], ["admin"]]
    )
    class Product(SomeInterface):
        upc: str
```



Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Tyger Taco for the PR 👏

Get it here 👉 https://strawberry.rocks/release/(next)
```

**Comment by patrick91** [comment]
I think we could go up to 2.7, since it is in the [changelog](https://www.apollographql.com/docs/federation/federation-versions#v27)!

Thanks for this, I'll review soon 😊

**Comment by patrick91** [comment]
@TygerTaco could allow me to push some commits to your fork? 😊 I wanted to update the federation compatibility library version

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+```
+    @strawberry.federation.type(
+        authenticated=True,
+        policy=[["client", "poweruser"], ["admin"]],
+        requires_scopes=[["client", "poweruser"], ["admin"]]
+    )
+    class Product(SomeInterface):
+        upc: str
+```
```
Comment (patrick91) [inline RELEASE.md:13]:
````suggestion
This release adds support for Apollo Federation v2.6 which includes the `@authenticated`, `@requiresScopes`, and `@policy` directives.

As usual, we have first class support for them in the `strawberry.federation` namespace, here's an example:

```python
@strawberry.federation.type(
    authenticated=True,
    policy=[["client", "poweruser"], ["admin"]],
    requires_scopes=[["client", "poweruser"], ["admin"]]
)
class Product(SomeInterface):
    upc: str
```
````

**Review by patrick91 — COMMENTED** [review]
This looks solid!

I think we can merge it after fixing the federation compatibility check 😊

It would be nice to bump it to 2.7 if you have time!

File: RELEASE.md (hunk 1 above)
Comment (TygerTaco) [reply RELEASE.md]: Thanks! I'll play with that test and I'll it get \@override updated to get to 2.7.  Probably will be sometime Monday.

File: strawberry/federation/field.py [hunk 2]
```diff
@@ -87,13 +94,16 @@ def field(
     name: Optional[str] = None,
     is_subscription: bool = False,
     description: Optional[str] = None,
+    authenticated: bool = False,
+    external: bool = False,
+    inaccessible: bool = False,
+    policy: Optional[List[List[Federation__Policy]]] = None,
```
Comment (patrick91) [inline strawberry/federation/field.py:100]: I think we should change this to 
```suggestion
    policy: Optional[List[List[str]]] = None,
```

`Federation__Policy` is a scalar, and at the moment the typing for them is broken 😊

File: RELEASE.md (hunk 1 above)
Comment (TygerTaco) [reply RELEASE.md]: update to support fed spec 2.7

File: strawberry/federation/field.py (hunk 2 above)
Comment (TygerTaco) [reply strawberry/federation/field.py]: done. Can you offer any suggestion on how to fix the failing benchmark tests? i can't repo that error on my local machine.

Comment (patrick91) [reply strawberry/federation/field.py]: we can ignore that test 😊

I'll review and merge this tonight or tomorrow, thank you so much!

**Review by patrick91 — APPROVED** [review]
Awesome! Thank you!

**Pull request merged**